lib/internal/errors.js

```diff
@@ -64,6 +65,16 @@ const prepareStackTrace = (globalThis, error, trace) => {
     return f(error, trace);
   }
 
+  for (let l = trace.length - 1; l >= 0; l--) {
```

Comment:
nit: the old `hideStackFrames()` implementation had the following comment:
```
// Make sure the most outer `hideStackFrames()` function is used.
```

We should keep a similar comment around this loop to make the logic of iteration and further `splice()` more obvious.

Reply:
Oh, yes, this was not intentional